source/common/router/context_impl.h

```diff
@@ -13,11 +13,14 @@ namespace Router {
 #define ALL_ROUTER_STATS(COUNTER, GAUGE, HISTOGRAM, TEXT_READOUT, STATNAME)                        \
   COUNTER(no_cluster)                                                                              \
   COUNTER(no_route)                                                                                \
+  COUNTER(passthrough_internal_redirect_stream_request)                                            \
+  COUNTER(passthrough_internal_redirect_body_handling)                                             \
   COUNTER(passthrough_internal_redirect_bad_location)                                              \
   COUNTER(passthrough_internal_redirect_no_route)                                                  \
   COUNTER(passthrough_internal_redirect_predicate)                                                 \
   COUNTER(passthrough_internal_redirect_too_many_redirects)                                        \
   COUNTER(passthrough_internal_redirect_unsafe_scheme)                                             \
+  COUNTER(passthrough_internal_redirect_recreate_stream_failed)                                    \
```

Comment:
Can you add this to a test? Seems like the other have a test associated with them.